main.go

```diff
@@ -106,24 +115,43 @@ var (
 )
 
 // scanDevices uses smartctl to gather the list of available devices.
-func scanDevices(logger log.Logger) []string {
+func scanDevices(logger log.Logger) []Device {
 	filter := newDeviceFilter(*smartctlDeviceExclude, *smartctlDeviceInclude)
 
 	json := readSMARTctlDevices(logger)
 	scanDevices := json.Get("devices").Array()
-	var scanDeviceResult []string
+	var scanDeviceResult []Device
 	for _, d := range scanDevices {
-		deviceName := d.Get("name").String()
+		deviceName := d.Get("info_name").String()
 		if filter.ignored(deviceName) {
 			level.Info(logger).Log("msg", "Ignoring device", "name", deviceName)
 		} else {
 			level.Info(logger).Log("msg", "Found device", "name", deviceName)
-			scanDeviceResult = append(scanDeviceResult, deviceName)
+			device := Device{
+				Name:      d.Get("name").String(),
+				Info_Name: d.Get("info_name").String(),
+				Protocol:  d.Get("protocol").String(),
```

Comment:
This specific copy of `Protocol` is written here and not read anywhere else in the codebase.
There's this related line over in `smartctl.go`, that reads it again from the smartctl output, where it could have read this field instead. 
```
smartctl.go:            protocol:   strings.TrimSpace(json.Get("device.protocol").String()),
```

If the Protocol is *NOT* needed in the type for other processing (like `device.Type` in readjson.go), why not drop it?